### PR TITLE
Offline warns + warns expiry 

### PR DESCRIPTION
**Pull request by ggggg**

### CODE REVIEW

File: src/BPEssentials/Configuration/Models/SettingsModel.cs [hunk 1]
```diff
@@ -81,4 +84,9 @@ public class General
 
         public bool LimitNames { get; set; }
     }
-}
+
+    public class warns
```
Comment (PLASMAchicken) [inline src/BPEssentials/Configuration/Models/SettingsModel.cs:88]: ....

File: src/BPEssentials/Core.cs [hunk 2]
```diff
@@ -159,6 +160,24 @@ public void RegisterCustomCommands()
...
+                foreach (var warn in warns)
+                {
+                    if (warn.Date.AddDays(warn.Length) <= DateTime.Now)
+                    {
+                        warns.Remove(warn);
+                    }
+                }
+            }
+        }
+
```
Comment (PLASMAchicken) [inline src/BPEssentials/Core.cs:180]: Should we delete them or mark them as expired?

File: src/BPEssentials/Configuration/Models/SettingsModel.cs [hunk 3]
```diff
@@ -16,6 +16,8 @@ public class Settings
 
         public ReportOptions ReportOptions { get; set; }
 
+        public warns Warns { get; set; }
```
Comment (PLASMAchicken) [inline src/BPEssentials/Configuration/Models/SettingsModel.cs:19]: ....

File: dist/settings.json [hunk 4]
```diff
@@ -18,6 +18,9 @@
     "KitMode": "Suggest",
     "SetJobMode": "Suggest"
   },
+  "Warns" : {
+      "DefaultWarnsExpirationInDays" : 20
+   }
```
Comment (PLASMAchicken) [inline dist/settings.json:23]:
```suggestion
   },
```

File: src/BPEssentials/Core.cs (hunk 2 above)
Comment (ggggg) [reply src/BPEssentials/Core.cs:180]: I can give that a setting

File: src/BPEssentials/Configuration/Models/SettingsModel.cs (hunk 1 above)
Comment (ggggg) [reply src/BPEssentials/Configuration/Models/SettingsModel.cs:88]: uh?

Comment (PLASMAchicken) [reply src/BPEssentials/Configuration/Models/SettingsModel.cs:88]: Warns

File: src/BPEssentials/ExtensionMethods/PlayerWarns.cs [hunk 5]
```diff
@@ -52,7 +64,7 @@ public static List<SerializableWarn> GetWarns(this ShPlayer player)
             {
                 warns = new List<SerializableWarn>();
             }
-            return warns;
+            return warns.FindAll(x => !x.Expired);
```
Comment (PLASMAchicken) [inline src/BPEssentials/ExtensionMethods/PlayerWarns.cs:67]: I don't think this should be done here.

**Pull request merged**